collector-sqs/src/main/java/zipkin/collector/awssqs/AwsSqsCollector.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright 2016 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.collector.awssqs;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.sqs.AmazonSQSAsync;
+import com.amazonaws.services.sqs.AmazonSQSAsyncClient;
+import com.amazonaws.services.sqs.buffered.AmazonSQSBufferedAsyncClient;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import zipkin.Component;
+import zipkin.collector.Collector;
+import zipkin.collector.CollectorComponent;
+import zipkin.collector.CollectorMetrics;
+import zipkin.collector.CollectorSampler;
+import zipkin.internal.LazyCloseable;
+import zipkin.internal.Util;
+import zipkin.storage.StorageComponent;
+
+
+public class AwsSqsCollector implements CollectorComponent, Closeable {
+
+  private static final Logger log = LoggerFactory.getLogger(AwsSqsCollector.class);
+
+  public static final class Builder implements CollectorComponent.Builder {
+
+    Collector.Builder delegate = Collector.builder(AwsSqsCollector.class);
+
+    List<String> queueUrls;
+    int waitTimeSeconds = 20; // aws sqs max wait time is 20 seconds
+    AWSCredentialsProvider credentialsProvider = new DefaultAWSCredentialsProviderChain();
+    int parallelism = 1;
+
+    @Override public CollectorComponent.Builder storage(StorageComponent storageComponent) {
+      delegate.storage(storageComponent);
+      return this;
+    }
+
+    @Override public CollectorComponent.Builder metrics(CollectorMetrics metrics) {
+      delegate.metrics(Util.checkNotNull(metrics, "metrics").forTransport("sqs"));
+      return this;
+    }
+
+    @Override public CollectorComponent.Builder sampler(CollectorSampler sampler) {
+      delegate.sampler(sampler);
+      return this;
+    }
+
+    /** SQS queue URLs to consume from */
+    public Builder queueUrls(List<String> queueUrls) {
```

Comment:
With infrastructure and queues in multiple regions, having the ability to consume from all the queues helps to consolidate resources.  This isn't something we don't strictly need but it would help eliminate extra resources, especially in lower traffic environments.

That said, after adding the multiple queues, I realized that the parallelism factor should be paired with a queue URL.  For our use case, having higher parallelism on certain queues would be ideal.

As an alternative I think it would be OK to make this a single queue and we will just run more instances of Zipkin pointing at different queues on the same EC2 instances.